addons/binding/org.openhab.binding.allplay/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,177 @@
+			<channel id="currentalbum" typeId="currentalbum" />
+			<channel id="currentartist" typeId="currentartist" />
+			<channel id="currentduration" typeId="currentduration" />
+			<channel id="currentgenre" typeId="currentgenre" />
+			<channel id="currenttitle" typeId="currenttitle" />
+			<channel id="currenturl" typeId="currenturl" />
+			<channel id="currentuserdata" typeId="currentuserdata" />
+			<channel id="loopmode" typeId="loopmode" />
+			<channel id="mute" typeId="mute" />
+			<channel id="playstate" typeId="playstate" />
+			<channel id="shufflemode" typeId="shufflemode" />
+			<channel id="stop" typeId="stop" />
+			<channel id="stream" typeId="stream" />
+			<channel id="volume" typeId="volume" />
+			<channel id="volumecontrol" typeId="volumecontrol" />
+			<channel id="zoneid" typeId="zoneid" />
+		</channels>
+
+		<config-description>
+			<parameter name="deviceId" type="text">
+				<label>Device ID</label>
+				<description>The device identifier identifies one certain speaker.</description>
+				<required>true</required>
+			</parameter>
+			<parameter name="deviceName" type="text">
+				<label>Device Name</label>
+				<description>The device name of the speaker.</description>
+				<required>false</required>
+			</parameter>
+			<parameter name="volumeStepSize" type="integer">
+				<label>Volume step size</label>
+				<description>Step size to use if the volume is changed using the increase/decrease command.</description>
+				<required>true</required>
+				<default>5</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<!-- Channel Type -->
+	<channel-type id="control">
+		<item-type>Player</item-type>
+		<label>Control</label>
+		<description>Control the AllPlay speaker, e.g. start/pause/next/previous/ffward/rewind</description>
+		<category>Player</category>
+	</channel-type>
+	<channel-type id="coverart">
+		<item-type>Image</item-type>
+		<label>Cover Art</label>
+		<description>Cover art image of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="coverarturl" advanced="true">
+		<item-type>String</item-type>
+		<label>Cover Art URL</label>
+		<description>Cover art URL of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currentalbum">
+		<item-type>String</item-type>
+		<label>Current Album</label>
+		<description>Album of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currentartist">
+		<item-type>String</item-type>
+		<label>Current Artist</label>
+		<description>Artist of the track currently playing</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="currentduration" advanced="true">
+		<item-type>Number</item-type>
+		<label>Current Duration</label>
+		<description>Duration in milliseconds of the track currently playing</description>
```

Comment:
> I would divide this by 1000 to get the seconds

Yes, but in the code and not leave it to (every) user to do it.

> or did you have something else in mind?

Yes, add `pattern="%d s"` to the channel-type, so that the value contains the unit when displayed in a UI.